test/mobile_app_backend_web/controllers/route_controller_test.exs

```diff
@@ -0,0 +1,58 @@
+defmodule MobileAppBackendWeb.RouteControllerTest do
+  use HttpStub.Case
+  use MobileAppBackendWeb.ConnCase
+  import Mox
+  import Test.Support.Helpers
+  import MobileAppBackend.Factory
+
+  describe "GET /api/route/stops unit tests" do
+    setup do
+      verify_on_exit!()
+      reassign_env(:mobile_app_backend, MBTAV3API.Repository, RepositoryMock)
+    end
+
+    defp mock_route_data do
+      andrew = build(:stop, id: "andrew")
+      jfk = build(:stop, id: "jfk/umass")
+      savin = build(:stop, id: "savin_hill")
+      north_quincy = build(:stop, id: "north_quincy")
+
+      RepositoryMock
+      |> expect(:stops, 1, fn params, _opts ->
+        case params
+             |> Keyword.get(:filter) do
+          [route: "Red", direction_id: "0"] ->
+            ok_response([andrew, jfk, savin, north_quincy])
+        end
+      end)
+
+      RepositoryMock
+      |> expect(:stops, 1, fn params, _opts ->
+        case params
+             |> Keyword.get(:filter) do
+          [route: "Red", direction_id: "1"] ->
+            ok_response([north_quincy, savin, jfk, andrew])
+        end
+      end)
```

Comment:
Point 4 in the [Mox docs](https://hexdocs.pm/mox/Mox.html) suggests that it’s more idiomatic to do something like
```elixir
RepositoryMock
|> expect(:stops, 2, fn
  [filter: [route: "Red", direction_id: "0"], fields: _], _opts -> ok_response([andrew, jfk, savin, north_quincy])
  [filter: [route: "Red", direction_id: "1"], fields: _], _opts -> ok_response([north_quincy, savin, jfk, andrew])
end)
```
rather than stacking multiple `expect` calls, although if this pattern works and we’re already doing it elsewhere then it’s probably fine.